### DISCUSSION TITLE
Whisper with CoreML support and speaker editing in the text editor.

### DISCUSSION THREAD

**Discussion by Jonik2** [kaixxx/noScribe#46] [Ideas]
First of all, I want to thank you for such an amazing tool, everything works beautifully and it's incredibly helpful in transcribing interviews.

I have three suggestions:

1) Is it possible to add Whisper with Core ML support for Macs on Apple Silicon? As far as I understand, this could significantly speed up the transcription process.

2) Is it possible to add the Confidence color-coding option (--print-colors) available in whisper.cpp? This could visually simplify the search for problematic areas in the transcription.

3) The text editor lacks one small feature to become the perfect editor for initial editing.
Sometimes short responses from the speaker are not identified, and I have to manually separate them into a separate line, for example:

**Original:**
S00 [00:00:27]: Then let's start. Of course. Please tell me more about your client.

**But it should be like this:**
S00 [00:00:27]: Then let's start.
S01 [00:00:28]: Of course.
S00 [00:00:29]: Please tell me more about your client.

Could you add an option to select text and press "change speaker"? This option would separate the selected text into a separate line as a phrase from another speaker, and assign it a timestamp from the file markup. Or simply by pressing "enter", a new paragraph would be formatted as the start of another speaker's speech.

**Comment by kaixxx** [comment]
Thanks for your suggestions!

> Whisper with Core ML support for Macs on Apple Silicon

We already use MPS with faster-whisper. Do you have something different in mind? I'm not an Apple-user...

> Is it possible to add the Confidence color-coding option

I had this in vers. 0.3 and found it more distracting than helpfull. You get a lot of false warnings, because most of the passages with low confidence are actually still correct. I've ditched the feature in vers. 0.4 and nobody complained (until now). But I might reintroduce it as an option, we will see.

> Could you add an option to select text and press "change speaker"?

That's a nice idea! I will put it on my list. I'm planning to add more functions to edit the speaker names semi-automatically in the editor.
